src/html_reporter.cc

```diff
@@ -0,0 +1,487 @@
+#include "benchmark/benchmark.h"
+#include "benchmark_api_internal.h"
+
+#include "complexity.h"
+#include "timers.h"
+
+#include <algorithm>
+#include <array>
+#include <cstdint>
+#include <iostream>
+#include <functional>
+#include <fstream>
+#include <string>
+#include <vector>
+
+
+static const char *const benchmark_titles[4] = {
+    "CPU time", "Items per second", "Real time", "Bytes per second"};
+
+static const char *const yaxis_titles[4] = {
+    "CPU time [ns]", "Items per second [items/s]", "Real time [ns]",
+    "Bytes per second [B/s]"};
+
+static const char *const line_tool_tip_bodies[4] = {
+    "Calculated values: <b>{point.key}</b><br/>CPU time: <b>{point.y} "
+    "ns</b><br/>Argument 1: <b>{point.x}</b>",
+    "Calculated values: <b>{point.key}</b><br/>Items per second: "
+    "<b>{point.y}</b><br/>Argument 1: <b>{point.x}</b>",
+    "Calculated values: <b>{point.key}</b><br/>Real time: <b>{point.y} "
+    "ns</b><br/>Argument 1: <b>{point.x}</b>",
+    "Calculated values: <b>{point.key}</b><br/>Bytes per second: "
+    "<b>{point.y}</b><br/>Argument 1: <b>{point.x}</b>"};
+
+static const char *const column_tool_tip_bodies[4] = {
+    "CPU time: <b>{point.y} ns</b>", "Items per second: <b>{point.y}</b>",
+    "Real time: <b>{point.y} ns</b>", "Bytes per second: <b>{point.y}</b>"};
+
+static const char *const high_chart_bar_function =
```

Comment:
examples like this are why i think having this be a separate service is the right approach. any changes to how the data should be rendered now require updates to the benchmark library. that's the wrong abstraction layer.